app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb

```diff
@@ -133,6 +135,41 @@ def hardware_networks(persister_hardware, instance)
     end
   end
 
+  def vm_and_template_labels(resource, tags)
+    tags.each do |tag|
+      formatted_tag = tag[:name].split(":")
+      tag_key = tag[:key] || formatted_tag[0] if formatted_tag.length >= 1
```

Comment:
This conditional isn't doing what you think, if `formatted_tag.length < 1` then `tag_key` will be `nil` because the if is applying to the whole line not just after the `||`

It is equivalent to saying
```
if formatted_tag.length >= 1
  tag_key = tag[:key] || formatted_tag[0]
end
```